chatkit-android/src/main/kotlin/com/pusher/chatkit/BeamsPushNotifications.kt

```diff
@@ -38,7 +32,25 @@ class BeamsPushNotifications(
     override fun setUserId(userId: String): Result<Unit, Error> {
         val f = FutureValue<Result<Unit, Error>>()
         try {
-            Beams.setUserId(userId, object : Callback<Void, PusherCallbackError> {
+            val tokenProvider = object : TokenProvider {
+                override fun fetchToken(userId: String): String {
```

Comment:
If this operation can fail, why was the return type changed from an optional?

Reply:
Because we decided to change the API so people explicitly throw exceptions instead of using optionals.

Return `null` doesn't explain _why_ you don't have a token. An exception does, which will lead to more meaningful errors.